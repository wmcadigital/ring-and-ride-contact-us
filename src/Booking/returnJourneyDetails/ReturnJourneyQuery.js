import { useEffect } from "react";
import PropTypes from "prop-types";
import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import ProgressIndicator from "../../common/ProgressIndicator";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutReturnJourneySectionName from "./getAboutReturnJourneySectionName";

import { required } from "../../common/validation";

const ReturnJourneyQuery = ({ setReturnJourney }) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const bookingParty = formValues["bookingParty"];

  const error = stateApi.submitFailed ? stateApi.errors?.returnJourney : null;

  const question =
    bookingParty === "behalfSomeone"
      ? `Does ${formValues["firstName"]} need to book a return journey on the same day?`
      : "Do you need to book a return journey on the same day?";

  useEffect(() => {
    if (formValues.returnJourney) {
      setReturnJourney(formValues.returnJourney);
    }
  }, [formValues.returnJourney]);

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("3", stateApi)}
        sectionName={getAboutReturnJourneySectionName(stateApi)}
      />
      <Question text={question} />
      <RadioGroup error={error}>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label="Yes"
          validation={required}
          value="yes"
          fieldName="returnJourney"
        />
        <RadioButton
          key={2}
          label="No"
          validation={required}
          value="no"
          fieldName="returnJourney"
        />
      </RadioGroup>
    </FormSection>
  );
};

ReturnJourneyQuery.propTypes = {
  setReturnJourney: PropTypes.func,
};

ReturnJourneyQuery.defaultProps = {
  setReturnJourney: () => {},
};

export default ReturnJourneyQuery;
