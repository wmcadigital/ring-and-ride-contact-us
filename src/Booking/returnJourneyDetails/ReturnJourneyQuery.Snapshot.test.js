import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import ReturnJourneyQuery from "./ReturnJourneyQuery";

jest.mock("react-final-form");

describe("ReturnJourneyQuery - snapshots", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<ReturnJourneyQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<ReturnJourneyQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for a group", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfGroup",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<ReturnJourneyQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
        firstName: "John",
      },
      submitFailed: true,
      errors: { returnJourney: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<ReturnJourneyQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
