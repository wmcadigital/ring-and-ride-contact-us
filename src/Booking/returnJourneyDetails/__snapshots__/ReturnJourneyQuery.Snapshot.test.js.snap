// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturnJourneyQuery - snapshots individual is filling out form for a group 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their return journey"
    sectionPosition="Section 4 of 5"
  />
  <Question
    text="Do you need to book a return journey on the same day?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="returnJourney"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="returnJourney"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`ReturnJourneyQuery - snapshots individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="Does John need to book a return journey on the same day?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="returnJourney"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="returnJourney"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`ReturnJourneyQuery - snapshots individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="Do you need to book a return journey on the same day?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="returnJourney"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="returnJourney"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`ReturnJourneyQuery - snapshots renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="Do you need to book a return journey on the same day?"
  />
  <RadioGroup
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="returnJourney"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="returnJourney"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;
